src/plugins/csvstorage/README.md

```diff
@@ -127,56 +127,6 @@ kdb rm -r user/tests/csv
 sudo kdb umount user/tests/csv
 ```
 
-# Directory Values
-
-By default the `csvstorage` plugin saves the name of the last column in each parent key. If you want to store a different value, you can do
-so using the [Directory Value](../directoryvalue/) plugin.
```

Comment:
Yep, this does not work anymore, since the `csvstorage` plugin does not add the meta key `array` to array parents. Before the pull request this part of the MSR test worked, since Directory Value only checked if the syntactic structure of a key set followed the array syntax. 

The example was not ideal anyway, since the plugin already stores the index of the last column element in the parent key and we just overwrite the value in the MSR test.

```sh
kdb mount config.csv user/tests/csvstorage csvstorage
printf 'one,two,three\nfour,five,six\n' > `kdb file user/tests/csvstorage`
kdb ls user/tests/csvstorage
#> user/tests/csvstorage/#0
#> user/tests/csvstorage/#0/#0
#> user/tests/csvstorage/#0/#1
#> user/tests/csvstorage/#0/#2
#> user/tests/csvstorage/#1
#> user/tests/csvstorage/#1/#0
#> user/tests/csvstorage/#1/#1
#> user/tests/csvstorage/#1/#2

# The plugin should add the array meta key to array parents, 
# but does not. This means the Directory Value plugin
# will not interpret `user/tests/csvstorage/#0` as array.
kdb meta-get user/tests/csvstorage/#0 array
# RET: 2

# Currently the plugin stores the index of the last column
# in the parent key.
kdb get user/tests/csvstorage/#0
#> 2
```

Anyway, the Directory Value plugin should behave correctly, if the storage plugin does so too. For example, both YAML CPP and YAJL should work. The `csvstorage` plugin does not, since it does not know who to serialize (write) a key such as `user/tests/csvstorage/#0/___dirdata` and `user/tests/csvstorage/#1/___dirdata` properly. These keys will be created by Directory Value, since the array metadata is missing. If we extend the example from before a little, then it looks like `csvstorage` currently stores the directory value (`___dirdata`) as last column in a file.

```sh
kdb set user/tests/csvstorage/#0 'Directory Value'
cat `kdb file user/tests/csvstorage`
#> one,two,three,Directory Value
#> four,five,six,#2
```

As a consequence setting the directory value will actually add an additional column to the CSV file.

```sh
kdb ls user/tests/csvstorage
#> user/tests/csvstorage/#0
#> user/tests/csvstorage/#0/#0
#> user/tests/csvstorage/#0/#1
#> user/tests/csvstorage/#0/#2
#> user/tests/csvstorage/#0/#3
#> user/tests/csvstorage/#1
#> user/tests/csvstorage/#1/#0
#> user/tests/csvstorage/#1/#1
#> user/tests/csvstorage/#1/#2
#> user/tests/csvstorage/#1/#3
```

Hopefully the description above explained the problem properly. Anyway, the easiest solution to fix the problem would probably be to add the meta key `array` to the parent keys. I tried that, but after an hour or so decided that the effort is not worth it for me, since no one will probably use this feature anyway.

Reply:
Thank you for the elaborate answer. I thought that the meta key `array` should suffice, but I fully agree that fixing this is not the highest priority.

Reply:
> I thought that the meta key `array` should suffice,…

That is correct. The `csvstorage` plugin does not add this meta key, hence setting the value of the directory key does not work.

Reply:
> Hopefully the description above explained the problem properly.

@sanssecours thank you very much! Can you please copy this to a separate issue? It would be difficult to find it here.

Reply:
> Can you please copy this to a separate issue?

I opened issue #3369, which contains a link to the [issue description above](https://github.com/ElektraInitiative/libelektra/pull/3357#discussion_r386127533).